src/pallene/pallenelib.lua

```diff
@@ -90,12 +127,209 @@ static lua_Number  pallene_math_log(lua_Integer x, lua_Integer base);
 static lua_Integer pallene_math_modf(lua_State *L, const char* file, int line, lua_Number n, lua_Number* out);
 
 /* Other builtins */
-static TString* pallene_string_char(lua_State *L, const char* file, int line, lua_Integer c);
-static TString* pallene_string_sub(lua_State *L, TString *str, lua_Integer start, lua_Integer end);
+static TString *pallene_string_char(lua_State *L, const char* file, int line, lua_Integer c);
+static TString *pallene_string_sub(lua_State *L, TString *str, lua_Integer start, lua_Integer end);
 static TString *pallene_type_builtin(lua_State *L, TValue v);
 static TString *pallene_tostring(lua_State *L, const char* file, int line, TValue v);
 static void pallene_io_write(lua_State *L, TString *str);
 
+/* Pallene tracer implementation. */
+
+static void pallene_tracer_frameenter(lua_State *L, pt_frame_t * restrict frame, ptrdiff_t sig) {
+    pt_frame_t *head, *tail;
+
+    lua_getglobal(L, "__pallene_tracer_stack_head");
+    head = (pt_frame_t *) lua_topointer(L, -1);
+    lua_getglobal(L, "__pallene_tracer_stack_tail");
+    tail = (pt_frame_t *) lua_topointer(L, -1);
+    lua_pop(L, 2);
+
+    /* If there is no frame in the stack. */
+    if(head == NULL) {
+        /* Just to be safe. */
+        frame -> prev = NULL;
+        frame -> next = NULL;
+
+        /* There is no other signature. */
+        frame -> frame_sig = sig;
+
+        head = frame;
+        tail = frame;
+
+        goto out;
+    }
+
+    /* The frame signature. */
+    /* If we don't have any frame signature, that denotes the call was made from Pallene
+       environment. */
+    frame -> frame_sig = sig == (ptrdiff_t) NULL ? tail -> frame_sig : sig;
```

Comment:
I found this hard to read without parenthesis. Maybe use an if statement instead?

Reply:
I find ternary cleaner and it's a little faster than if. I will try and make it more readable :).

Reply:
Performance won't be an issue!
Compilers generate identical assembly for both with `-O2`.

<img width="1508" alt="image" src="https://github.com/pallene-lang/pallene/assets/50487716/41fbec69-e104-4f66-81c8-e188ea2d2fc3">